test/test_allocation_domain.cpp

```diff
@@ -1168,4 +1168,31 @@ TEST_F(NVFuserTest, AllocationDomainContiguityForBroadcast) {
   testValidate(fusion, outputs, {t0}, {t1}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, AllocationDomainContiguityForExplicitBroadcast) {
```

Comment:
Is this related to your segmenter change?

Reply:
Don't remember pushing any segmenter changes recently....

But this comes from a python example for allocation domain when I was playing with my wip branch. :laughing:

Reply:
> Don't remember pushing any segmenter changes recently....

Isn't this PR changing segmenter?

Reply:
Sorry I got here late...

I think @zasdfgbnm means the above function `convertInputRfactorsToRoots` only applies at segmentation. So for example we could have a test like
```c++
auto tv1 = segment_set(tv0);
tv1->setAllocationDomain({alloc[1], alloc[0]});
auto tv2 = neg(tv1);
```
Then have check that after segmentation the input to the second segment expects transposed inputs.

Reply:
oh... sorry about this. I missed where the comment is at. :face_with_head_bandage: 

So the example here would trigger an assert on our existing code base. With the changes above the test passed with no problem.
In fact function `convertInputRfactorsToRoots` is called on all inputs even when no kernel segmentation happens.